spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -1388,6 +1388,19 @@ def compute_whitespace(line):
     @request(method=LSPRequestTypes.DOCUMENT_FOLDING_RANGE)
     def request_folding(self):
         """Request folding."""
+        total_lines = self.get_line_count()
+        if total_lines > 2000 and self.code_folding:
+            warn_seen = CONF.get('editor', 'code_folding_warn')
+            warn_str = _('This file contains more than 2000 lines! All '
+                         'code folding functionality will be disabled in '
+                         'order to prevent further performance degradation.')
+            if not warn_seen:
+                QMessageBox.information(self, _('File too long'), warn_str,
+                                        QMessageBox.Ok)
+                CONF.set('editor', 'code_folding_warn', True)
+            self.toggle_code_folding(False)

```

Comment:
What happens when you open a file with less than 2000 lines after this? It seems that folding is disabled for all files without considering their length?

Reply:
The code_folding flag is only disabled for that CodeEditor instance

Reply:
Ok, great! That's what I wanted to know!